node_modules/oae-transfer/lib/internal/dao.js

```diff
@@ -0,0 +1,778 @@
+
+                members.forEach(function(res, i, arrI) {
+
+                    // If the user is the manager and if the user isn't the user target
+                    if(res.role === "manager") {
+
+                        // Get the user
+                        Cassandra.runQuery('SELECT * FROM "Principals" WHERE "principalId" = ?', [res.id], function (err, manager) {
+                            if(err) {
+                                return callback(err);
+                            } 
+
+                            // Get resource
+                            _getDocument(ctx, hash.resourceId, function (err, element) {
+                                if(err) {
+                                    return callback(err);
+                                }
+
+                                // Insert to library
+                                _insertIntoLibrary(ctx, element, idUserTarget, principalIdOrigin, function (err, result) {
+                                    if(err) {
+                                        return callback(err);
+                                    }
+                                    
+                                    manager = _.chain(manager).map(Cassandra.rowToHash)._wrapped[0];
+                                    manager.resource = element;
+                                    managers.push(manager);
+                                    
+                                    // If all managers found, callback
+                                    if (arrI.length-1 === i) {
+                                        resolve(managers);
+                                    } 
+                                }); 
+                            }); 
+                        });
+                    } else {
+                        if (arrI.length-1 === i) {
+                            resolve(managers);
+                        } 
+                    }
+                });
+            });   
+        }); 
+    });
+
+    // Return managers
+    promise.then(function(managers) {
+        return callback(null, managers);
+    }).catch(function() {
+        return callback(null);
+    });
+};
+
+/**
+ * Get the name of the resource by it's id 
+ * take advantage of the differentiation of data type to modify the library
+ * 
+ * @param {Object} ctx
+ * @param {String} idResource
+ * @param {String} principalId
+ * @param {String} principalIdOrigin
+ *
+ * @returns nothing
+ * @api private
+ */
+var _insertIntoLibrary = function (ctx, resource, principalId, principalIdOrigin, callback) {
+    var resourceType = resource.id.split(":")[0];
+    
+    var constant = "";
+    var constantDiscussion = DiscussionsConstants.library.DISCUSSIONS_LIBRARY_INDEX_NAME;
+    var constantFolder = FoldersConstants.library.FOLDERS_LIBRARY_INDEX_NAME;
+    var constantContent = ContentConstants.library.CONTENT_LIBRARY_INDEX_NAME;
+    var constantMeeting = MeetingsConstants.library.MEETINGS_LIBRARY_INDEX_NAME;
```

Comment:
I'm not sure what benefit we get from reassigning these constants to variables here...